FreeRTOS/FreeRTOS-Plus-TCP
# Fix build warnings and update CI build checks to cover library header files

Description
-----------
This PR fixes warnings related to C standard compliance and updates the CMake files to include the +TCP library header files to be included in the build checks.

Updates the unit tests with respect to the changes made.

Test Steps
-----------
Fix incompatable code and build checks in CI.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
